front/ldap.import.php

```diff
@@ -40,6 +40,13 @@
 Session::checkRight("user", User::IMPORTEXTAUTHUSERS);
 
 // Need REQUEST to manage initial walues and posted ones
+if (isset($_REQUEST['basedn'])) {
+    $_REQUEST['basedn'] = Toolbox::stripslashes_deep($_REQUEST['basedn']);
+}
+if (isset($_REQUEST['ldap_filter'])) {
+    $_REQUEST['ldap_filter'] = Toolbox::stripslashes_deep($_REQUEST['ldap_filter']);
+}
```

Comment:
As filter may contains `&` char, you should use the `Sanitizer::unsanitize()` method to get the exact value from user input:
```suggestion
if (isset($_REQUEST['basedn'])) {
    $_REQUEST['basedn'] = Sanitizer::unsanitize($_REQUEST['basedn']);
}
if (isset($_REQUEST['ldap_filter'])) {
    $_REQUEST['ldap_filter'] = Sanitizer::unsanitize($_REQUEST['ldap_filter']);
}
```

will also require to add corresponding `use` statement and to patch display of the value:
```diff
diff --git a/src/AuthLDAP.php b/src/AuthLDAP.php
index f21e22ed9e..d300a8039b 100644
--- a/src/AuthLDAP.php
+++ b/src/AuthLDAP.php
@@ -3671,7 +3671,7 @@ class AuthLDAP extends CommonDBTM
                     }
 
                     echo "<tr><td style='width: 250px' class='text-end'><label for='basedn'>" . __('BaseDN') . "</label></td><td colspan='3'>";
-                    echo "<input type='text' class='form-control' id='basedn' name='basedn' value=\"" . $_SESSION['ldap_import']['basedn'] .
+                    echo "<input type='text' class='form-control' id='basedn' name='basedn' value=\"" . htmlspecialchars($_SESSION['ldap_import']['basedn'], ENT_QUOTES) .
                      "\" " . (!$_SESSION['ldap_import']['basedn'] ? "disabled" : "") . ">";
                     echo "</td></tr>";
```